src/MagnumPlugins/CgltfImporter/Test/CgltfImporterTest.cpp

```diff
@@ -0,0 +1,4435 @@
+        CORRADE_COMPARE(animation->trackType(1), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackResultType(1), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackTargetType(1), AnimationTrackTargetType::Translation3D);
+        CORRADE_COMPARE(animation->trackTarget(1), 1);
+        Animation::TrackView<const Float, const Vector3> translation = animation->track<Vector3>(1);
+        CORRADE_COMPARE(translation.interpolation(), Animation::Interpolation::Constant);
+        CORRADE_COMPARE(translation.before(), Animation::Extrapolation::Constant);
+        CORRADE_COMPARE(translation.after(), Animation::Extrapolation::Constant);
+        const Vector3 translationData[]{
+            Vector3::yAxis(0.0f),
+            Vector3::yAxis(2.5f),
+            Vector3::yAxis(2.5f),
+            Vector3::yAxis(0.0f)
+        };
+        CORRADE_COMPARE_AS(translation.keys(), Containers::stridedArrayView(translationScalingKeys), TestSuite::Compare::Container);
+        CORRADE_COMPARE_AS(translation.values(), Containers::stridedArrayView(translationData), TestSuite::Compare::Container);
+        CORRADE_COMPARE(translation.at(1.5f), Vector3::yAxis(2.5f));
+
+        /* Scaling, linearly interpolated, sharing keys with translation */
+        CORRADE_COMPARE(animation->trackType(2), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackResultType(2), AnimationTrackType::Vector3);
+        CORRADE_COMPARE(animation->trackTargetType(2), AnimationTrackTargetType::Scaling3D);
+        CORRADE_COMPARE(animation->trackTarget(2), 2);
+        Animation::TrackView<const Float, const Vector3> scaling = animation->track<Vector3>(2);
+        CORRADE_COMPARE(scaling.interpolation(), Animation::Interpolation::Linear);
+        CORRADE_COMPARE(scaling.before(), Animation::Extrapolation::Constant);
+        CORRADE_COMPARE(scaling.after(), Animation::Extrapolation::Constant);
+        const Vector3 scalingData[]{
+            Vector3{1.0f},
+            Vector3::zScale(5.0f),
+            Vector3::zScale(6.0f),
+            Vector3(1.0f),
+        };
+        CORRADE_COMPARE_AS(scaling.keys(), Containers::stridedArrayView(translationScalingKeys), TestSuite::Compare::Container);
+        CORRADE_COMPARE_AS(scaling.values(), Containers::stridedArrayView(scalingData), TestSuite::Compare::Container);
+        CORRADE_COMPARE(scaling.at(1.5f), Vector3::zScale(5.2f));
+    }
+}
+
+void CgltfImporterTest::animationOutOfBounds() {
+    auto&& data = AnimationOutOfBoundsData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("CgltfImporter");
+
+    std::ostringstream out;
+    Error redirectError{&out};
+    CORRADE_VERIFY(!importer->openFile(Utility::Directory::join(TINYGLTFIMPORTER_TEST_DIR, data.file)));
+    CORRADE_COMPARE(out.str(), "Trade::CgltfImporter::openData(): error opening file: invalid glTF\n");
```

Comment:
Would it make sense to expand on the generic error message a bit? Just to give people a hint on what *might be wrong*... Something like:

> "error opening file: invalid glTF, usually caused by invalid indices or missing required attributes"

Reply:
That would make sense indeed :+1: 

I'm afraid this could become a friction point, I remember people were already complaining that tinygltf's json.hpp only prints byte position but not line number when reporting JSON syntax errors, which makes debugging kinda annoying. If it becomes a problem, it'd probably have to be fixed on cgltf's side, not much to do here.

Reply:
We could probably link to the [glTF validator](https://github.khronos.org/glTF-Validator) in the docs? Just as a least-effort hint for now

Reply:
Good idea -- mention the possibility of vague errors in the behavior/limitations docs and add a link to the validator.